### PR TITLE
Improve GitHub clone/fork form

### LINKED ISSUE #6404: GitHub Plugin: Allow sorting on the column headers
## Feature description

When retrieving the "My repositories"/"Search for repositories" list via `GitHub` -> `Fork/Clone repositories`, the order in which the repos are listed has always been unfathomable to me.

See the red box in the screenshot for an example.

So, I'd like to suggest the following:

#### Basic solution

Sort the list of repositories received from the GH API by name (case-insensitive) before displaying them.

#### More advanced solution

Allow the end-user to sort the list based on the columns being display by clicking on any of the column headers (blue box in the screenshot).

Similar to other typical sorting interfaces in Windows programs, I would imagine this would work like so:
* One click on a column header: sort ascending on that column.
* Second click on that same column header: reverse sort order (sort descending)
* Click on another column header: sort ascending on that column with any previous sorting as secondary criteria.

GE-GH-sorting

## Environment

- Git Extensions 3.0.2.5232
- Build 23b6f51905006ccdde8701591df706284b4155dc
- Git 2.21.0.windows.1
- Microsoft Windows NT 6.1.7601 Service Pack 1
- .NET Framework 4.7.3362.0
- DPI 96dpi (no scaling)

### PR DESCRIPTION
**Pull request by pmiossec**
Fixes #6404

## Proposed changes

* Order results in the lists (My repositories & Search for repositories). Fixes #6404
* Improve column sizing and resizing
* Add option to easily call an upstream remote "upstream" (and update the label to be clearer)

### Before

image

### After

See:
* results sorted
* column width is better
* Editable dropdown to be able to select "upstream" + update on label
image

## Test methodology 

- Manual

## Test environment(s) 

- Git Extensions 3.2.0
- Build 8cc5933320a245b19c4fc7ff5a2285d319ac0d9e
- Git 2.21.0.windows.1
- Microsoft Windows NT 10.0.17134.0
- .NET Framework 4.7.3416.0
- DPI 192dpi (200% scaling)

### CODE REVIEW

**Comment by RussKie** [comment]
Would it be a big deal to center-align the 3 columns?

File: GitUI/CommandsDialogs/RepoHosting/ForkAndCloneForm.cs [hunk 1]
```diff
@@ -49,6 +51,16 @@ public ForkAndCloneForm(IRepositoryHostPlugin gitHoster, EventHandler<GitModuleE
             _gitHoster = gitHoster;
             InitializeComponent();
             InitializeComplete();
+
+            foreach (ColumnHeader column in myReposLV.Columns)
+            {
+                column.Width = (int)(column.Width * DpiUtil.ScaleX);
```
Comment (RussKie) [inline GitUI/CommandsDialogs/RepoHosting/ForkAndCloneForm.cs]:
```suggestion
                column.Width = DpiUtil.Scale(column.Width);
```

**Comment by pmiossec** [comment]
>Would it be a big deal to center-align the 3 columns?

No, that should not be a big deal. But don't you think that right aligned is better? Because center alignment for numbers is strange...

**Comment by RussKie** [comment]
True, though for "Yes"/"No" center is probably better
Can we reorder "Private" "Is fork" "# Forks" (does it make sense to re-order)?

**Comment by pmiossec** [comment]
>True, though for "Yes"/"No" center is probably better

Ok

> Can we reorder "Private" "Is fork" "# Forks" (does it make sense to re-order)?

No, they are not re-orderable. And, No. There is no added value to do it for something that is barely read...

**Comment by RussKie** [comment]
> No, they are not re-orderable.

I meant in the designer, not at runtime.


> no added value to do it for something that is barely read...

👍 works for me

**Comment by pmiossec** [comment]
I have made some more column adjustments:

* change column alignments (centered for yes/no and right for numbers)
* rename column headers
* change column order to have the same in the 2 lists
* change column header text to have the same in the 2 lists

Now, it looks like:
image


image

File: GitUI/Translation/English.xlf [hunk 2]
```diff
@@ -1557,7 +1557,7 @@ You will have push access. {2}</source>
         <target />
       </trans-unit>
       <trans-unit id="label3.Text">
-        <source>Add remote as:</source>
```
Comment (gerhardol) [inline GitUI/Translation/English.xlf]: Is this really the only translation related change?
I expected that the renamed controls and at least "Forks" -> "# Forks" to show here.

Comment (pmiossec) [reply GitUI/Translation/English.xlf]: Thanks for this remarks that make me see that some columns were not translated.
I fixed it.

**Comment by pmiossec** [comment]
Should I do something to move this PR forward?

**Comment by gerhardol** [comment]
> Should I do something to move this PR forward?

Same as for #6417, you should be able to merge yourself: Approved (including RussKie) and you have now announced the intention that would allow someone late to the party to review. Merge tomorrow.

**Comment by jrfnl** [comment]
@pmiossec Thanks for this. Changes look great! I look forward to seeing them in the next release!

**Pull request merged**